addons/mass_mailing/controllers/main.py

```diff
@@ -55,3 +87,51 @@ def full_url_redirect(self, code, stat_id, **post):
 
         request.env['link.tracker.click'].add_click(code, request.httprequest.remote_addr, country_code, stat_id=stat_id)
         return werkzeug.utils.redirect(request.env['link.tracker'].get_url_from_code(code), 301)
+
+    @http.route('/mail/mailing/blacklist/check', type='json', auth='none')
+    def check_blacklist(self, email):
+        if email:
+            record = request.env['mail.blacklist'].sudo().with_context(active_test=False).search([('email', '=ilike', email)])
```

Comment:
That's a security hole: I can find blacklisted people on your database as there is a sudo without any protection.